From a6e66cf9a8c76a12fc3aa6d4e307858f743eab0d Mon Sep 17 00:00:00 2001
From: Bogdan Purcareata <bogdan.purcareata@nxp.com>
Date: Thu, 12 Oct 2017 12:55:41 +0000
Subject: [PATCH 171/448] staging: fsl-dpaa2/eth: Use signed variables for buffer accounting

When comparing each channel buf_count with the refill threshold and max
buffer per channel thershold, there's an implicit conversion to unsigned
which leads to underflow errors when per channel buffers are consumed
too fast.

Convert all variables to use signed types.

Signed-off-by: Bogdan Purcareata <bogdan.purcareata@nxp.com>
---
 drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.h |    4 ++--
 1 files changed, 2 insertions(+), 2 deletions(-)

diff --git a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.h b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.h
index 45406b8..7b37aa4 100644
--- a/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.h
+++ b/drivers/staging/fsl-dpaa2/ethernet/dpaa2-eth.h
@@ -390,8 +390,8 @@ struct dpaa2_eth_priv {
 	u16 tx_qdid;
 	u16 rx_buf_align;
 	struct iommu_domain *iommu_domain;
-	u32 max_bufs_per_ch;
-	u32 refill_thresh;
+	int max_bufs_per_ch;
+	int refill_thresh;
 	bool has_xdp_prog;
 
 	void *cscn_mem;	/* Tx congestion notifications are written here */
-- 
1.7.1

